### PR TITLE
tests: add OasstApiClient contract tests

### LINKED ISSUE #205: Contract tests between API and Discord Bot
As per discussion in #180, the communication between the discord bot and API should be tested through a Contract test. This test should test that each call satisfies the OpenAPI schema inputs and endpoints. That way, changes to the endpoint contract will automatically be caught by these tests.

### PR DESCRIPTION
**Pull request by jack-michaud**
Fixes #205 

- Adds print_openapi_schema to backend
- Add start-mock-server.sh backend script
- Add contract tests for OasstApiClient (with test script)
  - Contract tests will automatically catch changes to the backend OpenAPI schema. 
- Add some additional type information and response model annotations to fill out the empty OpenAPI schema

Not in this PR but will complete in my next PR: adding a github workflow which automatically runs the tests for each PR and push.

### CODE REVIEW

File: backend/oasst_backend/api/v1/tasks.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def tasks_acknowledge_failure(
         raise OasstError("Failed to not acknowledge task.", OasstErrorCode.TASK_NACK_FAILED)
 
 
-@router.post("/interaction")
+@router.post("/interaction", response_model=protocol_schema.TaskDone)
```
Comment (andreaskoepf) [inline backend/oasst_backend/api/v1/tasks.py:207]: @yk note on how to specify multiple possible response_models: https://stackoverflow.com/questions/72919006/is-there-any-way-to-have-multiple-response-models-in-fastapi-openapi

File: discord-bot/bot/api_client.py [hunk 2]
```diff
@@ -89,23 +91,22 @@ async def fetch_random_task(
         logger.debug(f"Fetching random for user {user}")
         return await self.fetch_task(protocol_schema.TaskRequestType.random, user, collective)
 
-    async def ack_task(self, task_id: str | UUID, message_id: str):
+    async def ack_task(self, task_id: str | UUID, message_id: str) -> None:
         """Send an ACK for a task to the backend."""
         logger.debug(f"ACK task {task_id} with post {message_id}")
         req = protocol_schema.TaskAck(message_id=message_id)
-        return await self.post(f"/api/v1/tasks/{task_id}/ack", data=req.dict())
```
Comment (andreaskoepf) [inline discord-bot/bot/api_client.py:96]: Currently `tasks/{task_id}/ack` returns an empty dict. Probably we can change that to an empty 200 OK response too (which NACK currently does).

File: backend/print_openapi_schema.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+import json
+
+from main import app
```
Comment (andreaskoepf) [inline backend/print_openapi_schema.py:4]: Is it common to place such a file in the project? should we maybe have only one entry point and specify 'print-api-schema' as command-argument? Not sure what other projects normally do.

File: discord-bot/bot/api_client.py [hunk 4]
```diff
@@ -53,14 +53,16 @@ def __init__(self, backend_url: str, api_key: str):
             TaskType.done: protocol_schema.TaskDone,
         }
 
-    async def post(self, path: str, data: dict[str, t.Any]) -> dict[str, t.Any]:
+    async def post(self, path: str, data: dict[str, t.Any]) -> Optional[dict[str, t.Any]]:
         """Make a POST request to the backend."""
         logger.debug(f"POST {self.backend_url}{path} DATA: {data}")
         response = await self.session.post(f"{self.backend_url}{path}", json=data, headers={"X-API-Key": self.api_key})
         response.raise_for_status()
```
Comment (andreaskoepf) [inline discord-bot/bot/api_client.py:60]: Thanks for improving the client interface. We probably also want to decode the exception information at the client if it is a OasstError exception .. that is something for a separate PR I will open an issue for it.

**Review by andreaskoepf — APPROVED** [review]
Remaining question about the best location for the print-api call .. but overall approved.

File: backend/print_openapi_schema.py (hunk 3 above)
Comment (jack-michaud) [reply backend/print_openapi_schema.py:4]: I'll add a command. It's cleaner to not have a separate file.

File: discord-bot/bot/api_client.py (hunk 2 above)
Comment (jack-michaud) [reply discord-bot/bot/api_client.py:96]: 028c918

File: backend/print_openapi_schema.py (hunk 3 above)
Comment (jack-michaud) [reply backend/print_openapi_schema.py]: dda1f1f

**Pull request merged**